### PR TITLE
suggested template adjustments

### PR DESCRIPTION
**Pull request by kewalaka**
A number of suggestions

- remove "enable telemetry" from the default example, it is already set by default.
- add the required AVM interfaces for things like private endpoints, roles, locks etc.

This also includes some the previous suggestion regarding making the location an optional param.

I note that the line endings in the template are CRLF, when I run make docs & make fmt this makes for a messy commit, maybe it should be LF in the template tho?

more as a discussion point - I'm finding I'm repeating the same work each time i start... 🙂

### CODE REVIEW

File: main.privateendpoint.tf [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# TODO remove this code & var.private_endpoints if private link is not support.  Note it must be included in this module if it is supported.
+resource "azurerm_private_endpoint" "this" {
+  for_each                      = var.private_endpoints
+  name                          = each.value.name != null ? each.value.name : "pe-${var.name}"
+  location                      = each.value.location != null ? each.value.location : var.location
+  resource_group_name           = each.value.resource_group_name != null ? each.value.resource_group_name : var.resource_group_name
+  subnet_id                     = each.value.subnet_resource_id
+  custom_network_interface_name = each.value.network_interface_name
+
```
[the remark this answers is not in the document]
Comment (matt-FFFFFF) [reply main.privateendpoint.tf:9]: this was missing from the original but seems a good time to fix it

File: _header.md [hunk 2]
```diff
@@ -13,5 +13,6 @@ Things to do:
    1. AZURE_CLIENT_ID
    1. AZURE_TENANT_ID
    1. AZURE_SUBSCRIPTION_ID
+1. Search and update TODOs within the code and remove the TODO comments once complete.
```
[the remark this answers is not in the document]
Comment (kewalaka) [reply _header.md:16]: I don't think this is right, it isn't a subtask of creating the test environment, shouldn't be indented?

**Comment by kewalaka** [comment]
I've taken the liberty of running ```make fmt && make docs``` in WSL2 with the last commit, that changes the line encoding to LF.  Would be nice to have this as a precommit that is more cross platform?

**Comment by matt-FFFFFF** [comment]
> I've taken the liberty of running `make fmt && make docs` in WSL2 with the last commit, that changes the line encoding to LF. Would be nice to have this as a precommit that is more cross platform?

Thanks @kewalaka 

I only use WSL or MacOS - I'll look to add a .gitattributes for the Windows folk

```text
# Set the default behavior, in case people don't have core.autocrlf set.
* text=lf
```

**Review by matt-FFFFFF — APPROVED** [review]

**Pull request merged**